roles/manage_cnp/tasks/main.yml

```diff
@@ -0,0 +1,116 @@
+---
+# tasks file for role: manage_cnp
+# Options"
+# add-db, add-schema, add-role
+# drop-db, drop-schema, drop-role
+#- name: cnp_task not defined in 'supported_tasks' list
+#  ansible.builtin.fail:
+#    msg: cnp_task - "{{ cnp_task }}" is not in supported_tasks list "{{ supported_tasks }}"
+#  when: cnp_task not in supported_tasks
+- name: Validate cnp_task exist in supported_tasks list
+  ansible.builtin.fail:
+    #fail: msg="One of the defined cnp_task does not exist in the supported_tasks list"
+    msg: "cnp_task - {{ cnp_task | difference(supported_tasks) }} is not defined in supported_tasks"
+  when: cnp_task | difference(supported_tasks)
```

Comment:
I think this was where the loop through cnp_task was needed, I think that `cnp_task | difference(supported_tasks)` should yield the same result, but I am not sure if the logic works on lists and not just sets? 
Something along the lines of 
```
- name: Validate cnp_task exist in supported_tasks list
  ansible.builtin.fail:
    msg: "cnp_task - {{ task }} is not in supported tasks"
  when: task not in supported_tasks
  loop: "{{ cnp_task }}"
  loop_control:
    loop_var: task
```

Reply:
tested. it is currently working.